app/services/pii/password_encryptor.rb

```diff
@@ -6,7 +6,7 @@ def initialize
     end
 
     def encrypt(plaintext, user_access_key)
-      encrypted_key_maker.make(user_access_key)
+      encrypted_key_maker.make(user_access_key) unless user_access_key.made?
```

Comment:
There are 2 inter-related things going on in this PR:

* optimize `UserAccessKey` to avoid repeated calls to encrypt the internal `random_r` (the encrypted_key).
* fix the `UserAccessKey` caching in the `SessionEncryptor` so that Rails reload in development doesn't throw up.

The first issue should apply everywhere, esp in production, where we would like to avoid trips to the HSM if we can.

The second issue spawned this PR. I will try the duck-typing hack to see if that helps.